cmd/podman/shared/funcs.go

```diff
@@ -21,7 +22,10 @@ func GetAuthFile(authfile string) string {
 	if runtimeDir != "" {
 		return filepath.Join(runtimeDir, "containers/auth.json")
 	}
-	return ""
+	if !rootless.IsRootless() {
+		return fmt.Sprintf("/run/containers/%d/auth.json", rootless.GetRootlessUID())
+	}
+	return fmt.Sprintf("/run/user/%d/containers/auth.json", rootless.GetRootlessUID())
```

Comment:
@QiWang19 does this work like the Podman 1.4.4 examples that I just posted in my reply to this PR?  i.e.  root and no-sudo users hit line 28 and users who sudo hit line 26?  

I'm not sure where the `/run/container/{id}/auth.json` style is coming from, almost feels like a forgotten change in the code paths.

Reply:
I want non-root and no-sudo hit line 28, sudo and root hit line26

/run/container/{id}/auth.json ,  if it's root the  rootless.GetRootlessUID() returns 0.
```
sudo -s podman login (from tom) : /run/containers/0/auth.json
sudo podman login (from tom) : /run/containers/0/auth.json
```